experimental/packages/propagator-aws-xray-lambda/test/AWSXRayLambdaPropagator.test.ts

```diff
@@ -0,0 +1,201 @@
+      const existingContext: SpanContext = {
+        traceId: traceId,
+        spanId: spanId,
+        traceFlags: TraceFlags.SAMPLED,
+        isRemote: true,
+      };
+      const ctx = trace.setSpanContext(context.active(), existingContext);
+
+      process.env[AWSXRAY_TRACE_ID_ENV_VAR] =
+        'Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=1';
+
+      const xrayHeaderSpanId = '4635c8688f46eb5d';
+      const xrayHeaderTraceId = '48165d1ad6255f4cb1ddf9999ef6d536';
+      carrier[AWSXRAY_TRACE_ID_HEADER] =
+        'Root=1-48165d1a-d6255f4cb1ddf9999ef6d536;Parent=4635c8688f46eb5d;Sampled=1';
+
+      const extractedSpanContext = trace
+        .getSpan(
+          xrayLambdaPropagator.extract(ctx, carrier, defaultTextMapGetter)
+        )
+        ?.spanContext();
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: xrayHeaderTraceId,
+        spanId: xrayHeaderSpanId,
+        isRemote: true,
+        traceFlags: TraceFlags.SAMPLED,
+      });
+    });
+
+    it('should return env variable context if there is no active context but carrier also has xray headers', () => {
+      const xrayEnvSpanId = '53995c3f42cd8ad8';
+      const xrayEnvTraceId = '8a3c60f7d188f8fa79d48a391a778fa6';
+      process.env[AWSXRAY_TRACE_ID_ENV_VAR] =
+        'Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=1';
+
+      carrier[AWSXRAY_TRACE_ID_HEADER] =
+        'Root=1-48165d1a-d6255f4cb1ddf9999ef6d536;Parent=4635c8688f46eb5d;Sampled=1';
+
+      const extractedSpanContext = trace
+        .getSpan(
+          xrayLambdaPropagator.extract(
+            context.active(),
+            carrier,
+            defaultTextMapGetter
+          )
+        )
+        ?.spanContext();
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: xrayEnvTraceId,
+        spanId: xrayEnvSpanId,
+        isRemote: true,
+        traceFlags: TraceFlags.SAMPLED,
+      });
+    });
+
+    it('should return current context if one exists', () => {
+      delete process.env[AWSXRAY_TRACE_ID_ENV_VAR];
+
+      const xrayHeaderSpanId = '53995c3f42cd8ad8';
+      const xrayHeaderTraceId = '8a3c60f7d188f8fa79d48a391a778fa6';
+      carrier[AWSXRAY_TRACE_ID_HEADER] =
+        'Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=1';
+      const extractedSpanContext = trace
+        .getSpan(
+          xrayLambdaPropagator.extract(
+            ROOT_CONTEXT,
+            carrier,
+            defaultTextMapGetter
+          )
+        )
+        ?.spanContext();
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: xrayHeaderTraceId,
+        spanId: xrayHeaderSpanId,
+        isRemote: true,
+        traceFlags: TraceFlags.SAMPLED,
+      });
+    });
+  });
+
+  describe('.fields()', () => {
+    it('should return a field with AWS X-Ray Trace ID header', () => {
+      const expectedField = xrayLambdaPropagator.fields();
```

Comment:
nit: this is the actual list of fields, right? :thinking:

Reply:
Yes, this lambda propagator is used in place of the existing xray propagator (and only in the lambda context).